lib/spack/spack/fetch_strategy.py

```diff
@@ -780,9 +780,16 @@ def __init__(self, **kwargs):
 
     @property
     def git_version(self):
-        output = self.git('--version', output=str, error=str)
-        match = re.search(r'git version (\S+)', output)
-        return Version(match.group(1)) if match else None
+        return GitFetchStrategy.version_from_git(self.git)
+
+    @staticmethod
+    def version_from_git(git_exe):
+        """Given a git executable, return the version.
+        """
+        version_output = git_exe('--version', output=str)
+        m = re.search(r'^(?:git version\s+)([\w.-]+)', version_output)
+        vstring = m.group(1)
+        return Version(vstring)
```

Comment:
Can you clarify the purpose of these changes? Is it needed to reuse in the unit tests? When working on this I was wondering if it would be possible to directly use the `determine_version` attribute from the `git` package to further reduce duplication, but I wasn't sure how to do that.

Reply:
> Can you clarify the purpose of these changes? Is it needed to reuse in the unit tests?

If you're referring to the existence of a `version_from_git` function, then yes it is for reuse in the tests.

I could update the regex to match yours (yours is fine)

> I was wondering if it would be possible to directly use the determine_version attribute from the git package to further reduce duplication, but I wasn't sure how to do that.

I am also not sure, the API doesn't exactly match what the tests want to feed it (an `Executable` object vs. a path to an executable). Also while it would be nice to consolidate the two, IMO it would be nice not to depend on the package definition (in other words, maybe the `Git` package.py should be the "customer" in this case).

Reply:
Yeah, your regex looks a lot more complicated lol